onnx/defs/shape_inference.h

```diff
@@ -97,9 +97,12 @@ struct InferenceContext {
     // input-types of all inputs are known.
     return ((index < getNumInputs()) && (getInputType(index) != nullptr));
   }
-  virtual const TensorProto* getInputData(size_t index) const = 0;
   virtual size_t getNumOutputs() const = 0;
   virtual TypeProto* getOutputType(size_t index) = 0;
+  virtual bool hasOutput(size_t index) {
+    return (index < getNumOutputs() && (getOutputType(index) != nullptr));
+  }
+  virtual const TensorProto* getInputData(size_t index) const = 0;
```

Comment:
minor nit, for future reference: better to avoid moving irrelevant lines around (like above one): keeps the PR to just changed lines, helps reviewer, and when tracking historical changes (blame) in future

Reply:
Fixed, thanks